sycl/include/CL/sycl/detail/item_base.hpp

```diff
@@ -0,0 +1,76 @@
+//==---------- item_base.hpp --- SYCL iteration item_base ------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/id.hpp>
+#include <CL/sycl/range.hpp>
+
+namespace cl {
+namespace sycl {
+template <int dimensions> class id;
+template <int dimensions> class range;
+
+namespace detail {
+template <int dims, bool with_offset> struct item_base;
+
+template <int dims> struct item_base<dims, true> {
```

Comment:
Why not class?

Reply:
I don't need a class here, I have enough structure.
I don't use inheritance and fields of item_base  should be public.